internal/access/login.go

```diff
@@ -3,27 +3,10 @@ package access
 import (
 	"time"
 
-	"github.com/gin-gonic/gin"
-
 	"github.com/infrahq/infra/internal/server/authn"
-	"github.com/infrahq/infra/internal/server/models"
 )
 
 // Login uses a login method to authenticate a user
-func Login(c *gin.Context, loginMethod authn.LoginMethod, keyExpiresAt time.Time, keyExtension time.Duration) (*models.AccessKey, string, bool, error) {
-	db := getDB(c)
-	key, bearer, err := authn.Login(c.Request.Context(), db, loginMethod, keyExpiresAt, keyExtension)
-	if err != nil {
-		return nil, "", false, err
-	}
-
-	// In the case of username/password credentials,
-	// the login may fail if the password presented was a one-time password that has been used.
-	// This can be removed when #1441 is resolved
-	requiresUpdate, err := loginMethod.RequiresUpdate(db)
-	if err != nil {
-		return nil, "", false, err
-	}
-
-	return key, bearer, requiresUpdate, nil
+func Login(c RequestContext, loginMethod authn.LoginMethod, keyExpiresAt time.Time, keyExtension time.Duration) (authn.LoginResult, error) {
+	return authn.Login(c.Request.Context(), c.DBTxn, loginMethod, keyExpiresAt, keyExtension)
```

Comment:
I would really like to remove this `access.Login` function, but I suspect that is a controversial change.

If we remove this function we'd have the api handler calling the `authn.Login` directly, which removes one hop when tracing the code. The `access.Login` function does nothing except pull two values out of RequestContext, which could easily be done by the caller without adding additional lines.